cropper/src/main/java/com/canhub/cropper/CropImageContracts.kt

```diff
@@ -0,0 +1,483 @@
+import androidx.annotation.DrawableRes
+import com.canhub.cropper.CropImage.getActivityResult
+import com.canhub.cropper.CropImageView.CropShape
+import com.canhub.cropper.CropImageView.Guidelines
+import com.canhub.cropper.CropImageView.RequestSizeOptions
+
+class CropImageContract :
+    ActivityResultContract<CropImageContractOptions, CropImage.ActivityResult?>() {
+
+    override fun createIntent(context: Context, input: CropImageContractOptions): Intent {
+        input.options.validate()
+        return Intent().apply {
+            val bundle = Bundle()
+            bundle.putParcelable(CropImage.CROP_IMAGE_EXTRA_SOURCE, input.uri)
+            bundle.putParcelable(CropImage.CROP_IMAGE_EXTRA_OPTIONS, input.options)
+            putExtra(CropImage.CROP_IMAGE_EXTRA_BUNDLE, bundle)
+        }
+    }
+
+    override fun parseResult(
+        resultCode: Int,
+        intent: Intent?
+    ): CropImage.ActivityResult? {
+        return getActivityResult(intent)
+    }
+}
+
+fun options(
+    uri: Uri? = null,
+    builder: CropImageContractOptions.() -> (Unit) = {}
+): CropImageContractOptions {
+    val options = CropImageContractOptions(uri, CropImageOptions())
+    options.run(builder)
+    return options
+}
+
+data class CropImageContractOptions @JvmOverloads constructor(
+    val uri: Uri? = null,
+    val options: CropImageOptions = CropImageOptions()
+) {
+
+    /**
+     * The shape of the cropping window.<br></br>
+     * To set square/circle crop shape set aspect ratio to 1:1.<br></br>
+     * *Default: RECTANGLE*
+     *
+     * When setting RECTANGLE_VERTICAL_ONLY or RECTANGLE_HORIZONTAL_ONLY you may also want to
+     * use a free aspect ratio (to allow the crop window to change in the desired dimension
+     * whilst staying the same in the other dimension) and have the initial crop window cover
+     * the entire image (so that the crop window has no space to move in the other dimension).
+     * These can be done with
+     * [setFixAspectRatio] } (with argument `false`) and
+     * [setInitialCropWindowPaddingRatio] (with argument `0f).
+     */
+    fun setCropShape(cropShape: CropShape): CropImageContractOptions {
+        options.cropShape = cropShape
+        return this
+    }
+
+    /**
+     * An edge of the crop window will snap to the corresponding edge of a specified bounding box
+     * when the crop window edge is less than or equal to this distance (in pixels) away from the
+     * bounding box edge (in pixels).<br></br>
+     * *Default: 3dp*
+     */
+    fun setSnapRadius(snapRadius: Float): CropImageContractOptions {
+        options.snapRadius = snapRadius
+        return this
+    }
+
+    /**
+     * The radius of the touchable area around the handle (in pixels).<br></br>
+     * We are basing this value off of the recommended 48dp Rhythm.<br></br>
+     * See: http://developer.android.com/design/style/metrics-grids.html#48dp-rhythm<br></br>
+     * *Default: 48dp*
```

Comment:
I'm a little worry about the duplication of documentation between here and `CropImageOptions` 

yet, is very important to have in both places. Just not sure about the best way to maintain it. 

Just raising some concern, don't need to worry of making any change.

Reply:
This code is mostly copied from the `CropImage.ActivityBuilder`. I think the `ActivityBuilder` should be marked deprecated, and then the duplication can exist for a while, but in a future breaking release the old code is removed.

Reply:
like it! let's deprecate it